app/src/main/java/zapsolutions/zap/util/Wallet.java

```diff
@@ -194,6 +218,73 @@ public void run() {
         }
     }
 
+    /**
+     * Call this if the deamon is running, but the wallet is not unlocked yet.
+     *
+     * @param password
+     */
+    public void unlockWallet(String password) {
+        //UnlockWallet
+        WalletUnlockerGrpc.WalletUnlockerFutureStub asyncUnlockClient = WalletUnlockerGrpc
+                .newFutureStub(LndConnection.getInstance().getSecureChannel())
+                .withCallCredentials(LndConnection.getInstance().getMacaroon());
+
+        UnlockWalletRequest asyncUnlockRequest = UnlockWalletRequest.newBuilder()
+                .setWalletPassword(ByteString.copyFrom(password, Charset.defaultCharset()))
+                .build();
+
+        final ListenableFuture<UnlockWalletResponse> unlockFuture = asyncUnlockClient.unlockWallet(asyncUnlockRequest);
+
+        unlockFuture.addListener(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    UnlockWalletResponse unlockResponse = unlockFuture.get();
+                    ZapLog.debug(LOG_TAG, "successfully unlocked");
+
+                    if (PrefsUtil.isWalletSetup()) {
+                        LndConnection.getInstance().stopBackgroundTasks();
+                    }
+                    LndConnection.getInstance().restartBackgroundTasks();
+
+
+                    Handler handler = new Handler();
+                    handler.postDelayed(new Runnable() {
+                        @Override
+                        public void run() {
+                            // We have to call this delayed, as otherwise it will hang up lnd. (SubscribeToChannelEvents seems to cause this.)
+                            // To also get channels in an active state, this delay has to be quite high (5 seconds were not enough).
+                            isLNDReachable();
+                        }
+                    }, 1000);
```

Comment:
So I've tested this, and since its currently set to 1000ms, I get the problem that an error is shown prior to lnd is successfully unlocked. When I put in 5 seconds it seems to be okay.
There is no possibility to register to events like wallet unlocked before the wallet is actually unlocked?

I guess what we could do to workaround this issue is to call getinfo periodically every x ms, until a certain timeout (I think its 10 seconds currently) and check for a successful result?

Reply:
I have now increased it to 5 seconds as a temporary solution.
I don't think unlocking is a thing that is done often with zap android. 
So I would suggest we go with this for the time being and fix it with a periodically getinfo reqest in a separate PR.
See issue #48